src/Knp/DictionaryBundle/Dictionary/CallableDictionary.php

```diff
@@ -0,0 +1,162 @@
+<?php
+
+namespace Knp\DictionaryBundle\Dictionary;
+
+use Knp\DictionaryBundle\Dictionary as DictionaryInterface;
+
+class CallableDictionary implements DictionaryInterface
+{
+    /**
+     * @var string
+     */
+    private $name;
+
+    /**
+     * @var mixed[]|\ArrayAccess
+     */
+    private $values = null;
+
+    /**
+     * @var callable
+     */
+    private $callable;
+
+    /**
+     * @param string   $name
+     * @param callable $callable
+     */
+    public function __construct($name, $callable)
```

Comment:
With the current implementation we need to create a callable for every CallableDictionary we want to implement. In my case, I would like to pass a Repository method as the callable. But this method takes a single argument: the name of the list we would like to fetch.

So it would be cool to have a 3rd arguments, allowing to pass some arguments to the callable when values are resolved.